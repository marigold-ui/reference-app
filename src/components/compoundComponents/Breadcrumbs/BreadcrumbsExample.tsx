import Breadcrumbs from './Breadcrumbs';

function BreadcrumbsExample() {
  return (
    <>
      <Breadcrumbs>
        <Breadcrumbs.Crumb href="/">Electronics</Breadcrumbs.Crumb>
        <Breadcrumbs.Crumb href="/cellphones">Cell Phones</Breadcrumbs.Crumb>
        <Breadcrumbs.Crumb isCurrent={true} href="/cellphones/cases">
          Cases
        </Breadcrumbs.Crumb>
      </Breadcrumbs>
    </>
  );
}

export default BreadcrumbsExample;
