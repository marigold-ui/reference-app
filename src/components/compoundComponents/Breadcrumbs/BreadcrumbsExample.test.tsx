import { expect, test } from 'vitest';
import renderer from 'react-test-renderer';
import BreadcrumbsExample from './BreadcrumbsExample';

test('Breadcrumbs are correctly rendered', () => {
    const tree = renderer
        .create(<BreadcrumbsExample />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
