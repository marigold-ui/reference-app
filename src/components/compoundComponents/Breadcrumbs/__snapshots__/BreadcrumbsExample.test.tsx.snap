// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumbs are correctly rendered 1`] = `
<nav
  aria-label="Breadcrumbs"
>
  <ol
    className="m-0 list-none p-0"
  >
    <li
      className="mr-3 inline before:content-['/'] before:first-of-type:content-['']"
    >
      <a
        aria-current={false}
        className="text-decoration-none hover:text-decoration:revert ml-3 hover:underline"
        href="/"
      >
        Electronics
      </a>
    </li>
    <li
      className="mr-3 inline before:content-['/'] before:first-of-type:content-['']"
    >
      <a
        aria-current={false}
        className="text-decoration-none hover:text-decoration:revert ml-3 hover:underline"
        href="/cellphones"
      >
        Cell Phones
      </a>
    </li>
    <li
      className="mr-3 inline before:content-['/'] before:first-of-type:content-['']"
    >
      <a
        aria-current="page"
        className="text-decoration-none hover:text-decoration:revert ml-3 hover:underline"
        href="/cellphones/cases"
      >
        Cases
      </a>
    </li>
  </ol>
</nav>
`;
